libs/primitives/src/conversion.rs

```diff
@@ -0,0 +1,88 @@
+use sp_arithmetic::{
+	traits::{ensure_pow, BaseArithmetic},
+	ArithmeticError, FixedPointNumber,
+};
+
+/// Transform a fixed point number to a Balance.
+/// The resulting Balance will be represented with the `decimals` given.
+pub fn fixed_point_to_balance<
+	FixedPoint: FixedPointNumber<Inner = IntoBalance>,
+	IntoBalance: BaseArithmetic + Copy,
+>(
+	fixed_point: FixedPoint,
+	decimals: usize,
+) -> Result<IntoBalance, ArithmeticError> {
+	let integer_part = fixed_point.into_inner().ensure_div(FixedPoint::DIV)?;
+	let frac_part = fixed_point
+		.into_inner()
+		.ensure_sub(integer_part * FixedPoint::DIV)?;
+
+	let magnitude = ensure_pow(IntoBalance::from(10), decimals)?;
+
+	let new_integer_part = integer_part.ensure_mul(magnitude)?;
+	let new_frac_part = if magnitude > FixedPoint::DIV {
+		frac_part.ensure_mul(magnitude.ensure_div(FixedPoint::DIV)?)?
+	} else {
+		frac_part
+			.ensure_mul(magnitude)?
+			.ensure_div(FixedPoint::DIV)?
+	};
+
+	new_integer_part.ensure_add(new_frac_part)
```

Comment:
I know it's quite cryptic. But not sure how I can do it more "functional" because it's very low-level and mathematical. Maybe a comment about what math operations are there would help, and some examples. Ideally this kind of utility should be provided by `Substrate` or `orml` under some fixed_point module or similar.

`FixedPoint` is a DIV-decimal value. The amount of decimals depends on which implementation is used for `FixedPoint`.

Reply:
I've documented and added better examples to this conversion module. I hope it can be better understandable.